src/components/WorkOrder/Photos/hooks/uploadFiles/compressFile.ts

```diff
@@ -63,18 +49,21 @@ export async function compressFile(file: File): Promise<CompressResult> {
       compressResult = await compressWithBrowserImageCompression(file)
     }
   } catch (error) {
-    console.error('File compression failed:', error)
-    compressResult = {
-      success: false,
-      file: new File([file], file.name, {
-        type: file.type,
-        lastModified: file.lastModified,
-      }),
+    const compressLibrary = featureToggles.useCompressorJS ? 'CJS' : 'BIC'
+    let errorMessage = `File compression failed with ${compressLibrary} for "${file.name}":`
+    if (error instanceof ProgressEvent) {
+      // This error is thrown by browser-image-compression
+      const fileReaderError = (error.currentTarget as FileReader).error
+      errorMessage += ` ${fileReaderError}`
+      console.error(errorMessage)
+      throw new Error(errorMessage + ' Try re-uploading this file.')
     }
+    errorMessage += ` ${error instanceof Error ? error.message : error}`
+    console.error(errorMessage)
+    throw new Error(errorMessage)
```

Comment:
From testing, this returns the actual error messages thrown by the BIC and CJS libraries, instead of variations of [object Object] that we're currently getting in the logs